homeassistant/components/waze_travel_time/helpers.py

```diff
@@ -1,19 +1,25 @@
 """Helpers for Waze Travel Time integration."""
 import logging
 
-from WazeRouteCalculator import WazeRouteCalculator, WRCError
+from pywaze.route_calculator import WazeRouteCalculator, WRCError
 
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.httpx_client import get_async_client
 from homeassistant.helpers.location import find_coordinates
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def is_valid_config_entry(hass, origin, destination, region):
+async def is_valid_config_entry(
+    hass: HomeAssistant, origin: str, destination: str, region: str
+) -> bool:
     """Return whether the config entry data is valid."""
-    origin = find_coordinates(hass, origin)
-    destination = find_coordinates(hass, destination)
+    resolved_origin = find_coordinates(hass, origin)
+    resolved_destination = find_coordinates(hass, destination)
+    httpx_client = get_async_client(hass)
     try:
-        WazeRouteCalculator(origin, destination, region).calc_all_routes_info()
+        async with WazeRouteCalculator(region=region, client=httpx_client) as client:
```

Comment:
We shouldn't close the common Home Assistant httpx client session.

Reply:
done